ansible_collections/arista/avd/plugins/plugin_utils/avdfacts.py

```diff
@@ -17,15 +30,15 @@ def keys(cls):
         The "_" check is added to allow support for "internal" cached_properties storing temporary values.
         """
 
-        return [key for key in cls.__dict__ if not key.startswith("_") and isinstance(getattr(cls, key), cached_property)]
+        return [key for key in cls.__keys() if not key.startswith("_") and isinstance(getattr(cls, key), cached_property)]
```

Comment:
Nope. This will run on any instantiated class, where we want this to address keys of all the nested classes. Example with mixins, we want it to also cover the mixins. (That was the reason for doing this)